llama_index/llms/openai_utils.py

```diff
@@ -107,29 +110,45 @@
 CompletionClientType = Union[Type[Completion], Type[ChatCompletion]]
 
 
-def _create_retry_decorator(max_retries: int) -> Callable[[Any], Any]:
+def create_retry_decorator(
+    max_retries: int,
+    exponential: bool = True,
+    stop_after_delay_seconds: Union[int, None] = None,
+) -> Callable[[Any], Any]:
     min_seconds = 4
     max_seconds = 10
-    # Wait 2^x * 1 second between each retry starting with
-    # 4 seconds, then up to 10 seconds, then 10 seconds afterwards
+    wait_strategy = (
+        wait_exponential(multiplier=1, min=min_seconds, max=max_seconds)
+        if exponential
+        else wait_random_exponential(min=1, max=20)
+    )
+
+    stop_strategy: Union[stop_after_attempt, stop_any] = stop_after_attempt(max_retries)
```

Comment:
I think this should be `stop_strategy: stop_base`, not a `Union` of subclasses

Reply:
Okay 👌